admin/config-ui/components/class-component-post-type-visibility.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * @package WPSEO\Admin\ConfigurationUI
+ */
+
+/**
+ * Class WPSEO_Config_Component_Post_Type_Visibility
+ */
+class WPSEO_Config_Component_Post_Type_Visibility implements WPSEO_Config_Component {
+
+	/** @var array Option containing the data */
+	protected $option;
+
+	/**
+	 * Get the identifier
+	 *
+	 * @return string
+	 */
+	public function get_identifier() {
+		return 'PostTypeVisibility';
+	}
+
+	/**
+	 * Get the field to use
+	 *
+	 * @return WPSEO_Config_Field
+	 */
+	public function get_field() {
+		return new WPSEO_Config_Field_Post_Type_Visibility();
+	}
+
+	/**
+	 * Fetch the data
+	 *
+	 * @return mixed
+	 */
+	public function get_data() {
+
+		$this->option = WPSEO_Options::get_option( 'wpseo_xml' );
+
+		$post_types = get_post_types( array( 'public' => true ), 'names' );
+		array_walk( $post_types, array( $this, 'get_option_value' ) );
+
+		return $post_types;
+	}
+
+	/**
+	 * Save the data provided
+	 *
+	 * @param array $data Data provided containing changes.
+	 *
+	 * @return mixed
+	 */
+	public function set_data( $data ) {
+
+		$option = WPSEO_Options::get_option( 'wpseo_xml' );
+
+		foreach ( $data as $post_type => $in_sitemap ) {
+			$option[ 'post_types-' . $post_type . '-not_in_sitemap' ] = ( ! $in_sitemap );
```

Comment:
You're assuming that all post types are always sent as data to `set_data`. This is not a safe assumption.